pkg/interval/intervalset.go

```diff
@@ -38,9 +38,7 @@ func (c *CanonicalSet) Equal(other *CanonicalSet) bool {
 	return true
 }
 
-// AddInterval adds a new interval range to the set
-func (c *CanonicalSet) AddInterval(v Interval) {
-	set := c.IntervalSet
+func addInterval(v Interval, set []Interval) []Interval {
```

Comment:
What is the motivation behind this change?
Why have 2 `addInterval` functions when you can have one?

Reply:
Just an experiment. Reverted.